File: src/main/java/com/example/blog/entity/Article.java
package com.example.blog.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
@Table(name = "articles")
@Getter
@Setter
public class Article extends BaseEntity {

    @Column(nullable = false, length = 200)
    private String title;

    @Column(length = 255)
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String summary;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String content;

    @Column(length = 255)
    private String coverImage;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private Status status = Status.DRAFT;

    @Column(nullable = false)
    private Boolean isTop = false;

    @Column(nullable = false)
    private Boolean allowComment = true;

    @Column(nullable = false)
    private Integer viewCount = 0;

    @Column(nullable = false)
    private Integer likeCount = 0;

    private LocalDateTime publishedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "author_id", nullable = false)
    private User author;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
            name = "article_tags",
            joinColumns = @JoinColumn(name = "article_id"),
            inverseJoinColumns = @JoinColumn(name = "tag_id")
    )
    private Set<Tag> tags = new HashSet<>();

    @OneToMany(mappedBy = "article", fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    private List<Comment> comments = new ArrayList<>();

    public enum Status {
        DRAFT, PUBLISHED, ARCHIVED
    }
}

File: frontend/packages/shared/src/types/article.ts
export type ArticleStatus = 'DRAFT' | 'PUBLISHED' | 'ARCHIVED'

export interface AuthorInfo {
  id: number
  username: string
  nickname: string | null
  avatar: string | null
}

export interface ArticleDTO {
  id: number
  title: string
  slug: string
  summary: string | null
  content: string
  coverImage: string | null
  status: ArticleStatus
  isTop: boolean
  allowComment: boolean
  viewCount: number
  likeCount: number
  publishedAt: string | null
  author: AuthorInfo
  category: import('./category').CategoryDTO | null
  tags: import('./tag').TagDTO[]
  commentCount: number
  createdAt: string
  updatedAt: string
}

export interface ArticleRequest {
  title: string
  slug?: string
  summary?: string
  content: string
  coverImage?: string
  status?: ArticleStatus
  isTop?: boolean
  allowComment?: boolean
  categoryId?: number
  tagIds?: number[]
}
